lua/autorun/server/sv_autoclean.lua

```diff
@@ -1,36 +1,45 @@
-	CreateConVar('cfc_autoclean','500',{FCVAR_ARCHIVE, FCVAR_NOTIFY },'Autocleans the server based on seconds given')
-	
-function cfcCleanServer()
-	local player_weps = {}
-	local turrets = table.Add(ents.FindByClass("gmod_wire_turret"),ents.FindByClass("gmod_turret"))
-	local count = 0
-	--Manual fix for removing weapons held by player
-	for _ , v in pairs(player.GetHumans()) do
-		v:ChatPrint('[CFC_Autoclean] Cleaning server..')
-		--Also cleanup decals and sounds while we're here
-		v:ConCommand('r_cleardecals')
-		v:ConCommand('stopsound')
-
-		--
-		local wp = v:GetWeapons()
-		for _,m in pairs(wp) do
-			if (not table.HasValue(player_weps, m))then
-				table.insert(player_weps, m)
-			end
+
+local ConVarFlags = {FCVAR_ARCHIVE, FCVAR_NOTIFY}
+local DefaultCleanIntervalInSeconds = "500"	
+CreateConVar("cfc_autoclean", DefaultCleanIntervalInSeconds, ConVarFlags, "Autocleans the server based on seconds given")
+
+local BlacklistedEntityTypes = {
+	["gmod_wire_turret"] = true,
+	["gmod_turret"] = true
+}
+
+local CleanupCommands = {
+	["r_cleardecals"] = true,
+	["stopsound"] = true
+}
+
+local function runCleanupCommandsOnPlayers()
+	if not IsValid( ply ) then return end
+
+	for k,v in pairs( player.GetHumans() ) do
+		for command, _ in pairs( CleanupCommands ) do
+			ply:ConCommand( command )
 		end
 	end
+end
+
+function cfcCleanServer()
+	local removedCount = 0
 	
-	for k, v in pairs( ents.GetAll() ) do
-			if v:IsWeapon() and not (table.HasValue(player_weps,v) or (table.HasValue(turrets,v))) then
-				count = count + 1
-				v:Remove()
-			end
+	runCleanupCommandsOnPlayers()
+
+	for _, entity in pairs( ents.GetAll() ) do
+		if not IsValid( entity ) then continue end
```

Comment:
Why not 
```lua 
if not ( IsValid( entity ) and IsValid( entity.Owner ) ) then return end

local classIsBlacklisted = BlacklistedEntityTypes[entity:GetClass()]

if classIsBlacklisted or entity:isWeapon() then continue end

Reply:
ENT.Owner only works for weapons IIRC

Reply:
And the way you suggested just checks if it is a weapon then skips it, it wont remove grounded weapons.
ENT.Owner gets the current wielder of the weapon.

Reply:
Okay, here's another stab at it, then.

Also, I remember something about lua/glua not having the increment operators `+=` or `++`.

Can you look into that?

```lua
if not IsValid( entity ) then continue end

local classIsBlacklisted = BlacklistedEntityTypes[entity:GetClass()]
local isOwnedWeapon = entity:IsWeapon() and IsValid( entity.Owner )

if isOwnedWeapon or not classIsBlacklisted then continue end

entity:Remove()

removedEntities = removedEntities + 1
```

Reply:
I can do that for sure